File: Brevet_Evolution/src/Personnage/Personnage.java
/*
 * CC BY Project Brevet Evolution.
 *
 * Copyright CC BY 2016.
 * Brillet Camille.
 * Fillon Aurélien.
 * Foissac Nicola.
 * Hanotel Guillaume.
 * Henry-Brûlé Raphaël.
 * Marquay Christian.
 *
 * Creative Commons Attribution 3.0 France.
 * http://creativecommons.org/licenses/by/3.0/fr/legalcode
 *
 */
package Personnage;

/**
 *
 * @author nfoissac
 */
abstract public class Personnage
{

    private int vie;
    private String nom;

    public abstract String affichageVie();

    public abstract String affichageNom();
    
    public abstract boolean estMort();
    
    public abstract int prendDegats(int val);
    
    public int getVie()
    {
        return this.vie;
    }
    
    public String getNom()
    {
        return this.nom;
    }
    
    public void setVie(int val)
    {
        this.vie = val;
    }
    
    public void setNom(String nom)
    {
        this.nom = nom;
    }

}

File: fonctions_php/Personnage/Personnage.php
<?php
/*
 * CC BY Project Brevet Evolution.
 *
 * Copyright CC BY 2016.
 * Brillet Camille.
 * Fillon Aurélien.
 * Foissac Nicola.
 * Hanotel Guillaume.
 * Henry-Brûlé Raphaël.
 * Marquay Christian.
 *
 * Creative Commons Attribution 3.0 France.
 * http://creativecommons.org/licenses/by/3.0/fr/legalcode
 *
 */

/**
 *
 * @author nfoissac
 * @author cbrillet
 */
abstract class Personnage
{

    private $_vie;
    private $_nom;

    abstract public function affichageVie();

    abstract public function affichageNom();
    
    abstract public function estMort();
    
    abstract public function prendDegats($val);
    
    public function getVie()
    {
        return $this->_vie;
    }
    
    public function getNom()
    {
        return $this->_nom;
    }
    
    public function setVie($val)
    {
        $this->_vie = $val;
    }
    
    public function setNom($nom)
    {
        $this->_nom = $nom;
    }

}

?>
